Hippo.Web/Controllers/ClusterAdminController.cs

```diff
@@ -0,0 +1,134 @@
+using System;
+using System.Linq;
+using Hippo.Core.Data;
+using Hippo.Core.Models;
+using Hippo.Core.Services;
+using Hippo.Web.Models;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+
+namespace Hippo.Web.Controllers
+{
+    [Authorize(Policy = AccessCodes.SystemAccess)]
+    public class ClusterAdminController : Controller
+    {
+        private readonly ISecretsService _secretsService;
+        private readonly AppDbContext _dbContext;
+
+        public ClusterAdminController(ISecretsService secretsService, AppDbContext dbContext)
+        {
+            _secretsService = secretsService;
+            _dbContext = dbContext;
+        }
+
+        public IActionResult Index()
+        {
+            return View("React");
+        }
+
+        [HttpGet]
+        public async Task<IActionResult> Clusters()
+        {
+            var currentSecretKeyIds = new HashSet<string>(await _secretsService.GetSecretNames());
+
+            var clusterModels = (await _dbContext.Clusters
+                .AsNoTracking()
+                .OrderBy(c => c.Name)
+                .ToArrayAsync())
+                .Select(c => new ClusterModel(c, string.Empty))
+                .ToArray();
+
+            foreach (var model in clusterModels)
+            {
+                if (!string.IsNullOrEmpty(model.Cluster.SshKeyId) && currentSecretKeyIds.Contains(model.Cluster.SshKeyId))
+                {
+                    model.SshKey = await _secretsService.GetSecret(model.Cluster.SshKeyId);
```

Comment:
why do we need to get the actual key for each cluster?  shouldn't we only ever get the key at the moment we are going to push a file via ssh?